### PR TITLE
feat: [#186620631] add display name for funding opps

### PR DESCRIPTION
**Pull request by AnJuHyppolite**
## Description

This story adds/updates the following:

1. the funding opportunities in the CMS now has a `displayName` field
2. the summary has the agency id
3. the Content Team will now have the option to sort by agency
4. the id field is now read-only

### Ticket

[#186620631](https://www.pivotaltracker.com/story/show/186620631)

### Steps to Test

Access the funding opportunities in the CMS to visually see and test the updates.

## Code author checklist

- [x] I have performed a self-review of my code
- [ ] I have created and/or updated relevant documentation (EX: [Engineering Reference/FAQ](https://docs.google.com/document/d/1X4iWSGmBZdHYZ0jGqkwTR3_yyyqI_TiiJptfc8pi9Po)), if necessary
- [x] I have not used any relative imports
- [x] I have checked for and removed instances of unused config from CMS
- [x] I have pruned any instances of unused code
- [x] I have not added any markdown to labels, titles and button text in config
- [x] If I added/updated any values in `userData` (including `profileData`, `formationData` etc), then I added a new migration file
- [x] If I added any new collections to the CMS config, then I updated the search tool and `cmsCollections.ts`

### CODE REVIEW

File: web/public/mgmt/config.yml [hunk 1]
```diff
@@ -1233,19 +1233,21 @@ collections:
   - name: "funding-opportunities"
     label: "Fund Opps - Content"
     folder: "content/src/fundings"
-    summary: "/{{urlSlug}} - {{name}}"
+    summary: "[{{agency[0]}}] - {{name}}"
     identifier_field: "{{filename}}"
     slug: "{{filename}}"
     delete: true
+    sortable_fields: ["name", "agency"]
```
Comment (FarazA22) [inline web/public/mgmt/config.yml:1240]: dust - should we use displayName instead of name?

File: web/public/mgmt/config.yml [hunk 2]
```diff
@@ -1233,19 +1233,21 @@ collections:
   - name: "funding-opportunities"
     label: "Fund Opps - Content"
     folder: "content/src/fundings"
-    summary: "/{{urlSlug}} - {{name}}"
+    summary: "[{{agency[0]}}] - {{name}}"
```
Comment (FarazA22) [inline web/public/mgmt/config.yml:1236]: dust - should we use displayName instead of name?

File: content/src/fundings/dep-water-bank.md [hunk 3]
```diff
@@ -37,5 +38,5 @@ descriptionMd: This program provides low-cost financing for the design,
 - **Drinking water applicants** can be public community water systems privately and publicly owned, and nonprofit non-community water systems (as defined by the [National Primary Drinking Water Regulations](https://www.epa.gov/ground-water-and-drinking-water/national-primary-drinking-water-regulations))
 - **All applicants** must also satisfy the [Water Bank's creditworthiness standards](https://cdn.njib.gov/njib/policies/njib_credit_policy_2018.pdf)
 
-> ### Benefits
+> **Benefits:**
```
Comment (adidner) [inline content/src/fundings/dep-water-bank.md:41]: dust - I'm assuming this is an IDE change and these are equivalent

File: web/public/mgmt/config.yml (hunk 2 above)
Comment (adidner) [reply web/public/mgmt/config.yml:1236]: I second that, I think this should be displayname

File: web/public/mgmt/config.yml (hunk 1 above)
Comment (adidner) [reply web/public/mgmt/config.yml:1240]: I second that

File: web/public/mgmt/config.yml [hunk 4]
```diff
@@ -1233,19 +1233,21 @@ collections:
...
     fields:
       - { label: "Internal Notes", name: "notesMd", widget: "markdown", required: false }
       - { label: "File Name", name: "filename", widget: "write-once-read-only" }
       - { label: "Name", name: "name", widget: "string" }
+      - { label: "Display Name", name: "displayName", widget: "string" }
       - { label: "Url Slug", name: "urlSlug", widget: "string" }
-      - { label: "Id", name: "id", widget: "string" }
+      - { label: "Id", name: "id", widget: "write-once-read-only" }
       - {
           label: "Open Date",
```
Comment (adidner) [inline web/public/mgmt/config.yml:1252]: dust - I know we have Archieved fundings as well I assume this are being ignored / content doesn't care about them?

File: web/public/mgmt/config.yml (hunk 2 above)
Comment (adidner) [inline web/public/mgmt/config.yml:1236]: Never seen that syntax but played with it a bit and it seems neat.

Comment (adidner) [reply web/public/mgmt/config.yml:1236]: Would love it if we could somehow map to the agency name we typically use but I'm not sure if thats possible. So probably not worth the effort at the moment at least

File: web/public/mgmt/config.yml (hunk 1 above)
Comment (AnJuHyppolite) [reply web/public/mgmt/config.yml:1240]: Thanks for catching this!

File: content/src/fundings/dep-water-bank.md (hunk 3 above)
Comment (AnJuHyppolite) [reply content/src/fundings/dep-water-bank.md:41]: No. Merge conflict issues. I changed it back to `### Benefits`. Thank you!

File: web/public/mgmt/config.yml (hunk 4 above)
Comment (AnJuHyppolite) [reply web/public/mgmt/config.yml:1252]: I asked Content about the archived opps and they said not to include them.

**Pull request merged**